src/Scaffold_Command.php

```diff
@@ -201,6 +200,106 @@ private function _scaffold( $slug, $assoc_args, $defaults, $subdir, $templates )
 		}
 	}
 
+	/**
+	 * Generate PHP, JS and CSS code for registering a custom block.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <slug>
+	 * : The internal name of the block.
+	 *
+	 * [--title=<title>]
+	 * : The display title for your block.
+	 *
+	 * [--dashicon=<dashicon>]
+	 * : The dashicon to make it easier to identify your block.
+	 *
+	 * [--category=<category>]
+	 * : The category name to help users browse and discover your block.
+	 *
+	 * [--textdomain=<textdomain>]
+	 * : The textdomain to use for the labels.
+	 *
+	 * [--theme]
+	 * : Create files in the active theme directory. Specify a theme
+	 * with `--theme=<theme>` to have the file placed in that theme.
+	 *
+	 * [--plugin=<plugin>]
+	 * : Create files in the given plugin's directory.
+	 *
+	 * [--force]
+	 * : Overwrite files that already exist.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Generate a 'movie' block for the 'simple-life' theme
+	 *     $ wp scaffold block movie --title="Movie block" --theme=simple-life
+	 *      Success: Created block 'Movie block'.
+	 *
+	 * @subcommand block
+	 */
+	public function block( $args, $assoc_args ) {
+
+		$registered_categories = array(
+			'common',
+			'formatting',
+			'layout',
+			'widgets',
+			'embed',
+			'reusable-blocks',
+		);
+
+		$slug = $args[0];
+		if ( ! preg_match( '/^[a-z][a-z0-9\-]*$/', $slug ) ) {
+			WP_CLI::error( "Invalid block slug specified. Block slugs can contain only lowercase alphanumeric characters or dashes, and start with a letter." );
+		}
+
+		$defaults = array(
+			'title'      => str_replace( '-', ' ', $slug ),
+			'category'   => 'widgets',
+			'textdomain' => '',
+		);
+		$data     = $this->extract_args( $assoc_args, $defaults );
+
+		if ( ! in_array( $data['category'], $registered_categories ) ) {
+			WP_CLI::error( "Invalid block category specified. Block categories need to match one of the registered values." );
```

Comment:
This can be handled by adding YAML doc to the argument definition:

```
[--category=<category>]
: The category name to help users browse and discover your block.
---
options:
  - common
  - etc.
```

Reply:
Nice feature. I will use YAML instead 👍